### PR TITLE
chore: added workspace publishing docs

### PR DESCRIPTION
**Pull request by iakovn**
This PR adds a documentation section with an example on workspace publishing to cloud storage and importing the workspace back into userspace.

### CODE REVIEW

File: docs/source/working_with_workspaces.rst [hunk 1]
```diff
@@ -57,9 +57,48 @@ would result in multiple workspaces being created with same name.
...
+   published_workspace = pwc.find(name="PublishDemo")[0]
+   imported_workspace = published_workspace.import_to_userspace()
+
+The imported workspace will contain results that were bundled and experiment IDs will be kept intact::
+
+   imported_experiment = imported_workspace.get_experiments()[0]
+   assert imported_experiment.id == experiment.id
+
+Repeated imports of the same workspace do not update the workspace in userspace. However,
+If update was published receiver can get his copy overwritten by the update::
```
Comment (abhilash-kumar-nair) [inline docs/source/working_with_workspaces.rst:97]: If -> if

File: docs/source/working_with_workspaces.rst [hunk 2]
```diff
@@ -57,9 +57,48 @@ would result in multiple workspaces being created with same name.
...
 
-A workspace in Modelon Impact can be shared as a zipped artifact(snapshot) or as workspace definition referring to version controlled 
+A workspace in Modelon Impact can be shared as a zipped artifact (snapshot) or as a workspace configuration.
+A snapshot can be either stored in a cloud storage and shared with the other users of the same 
+Modelon Impact site or downloaded as a zip file to the client system.
+Workspace configuration is a JSON-serializable structure containing links to version controlled 
 projects and preinstalled libraries. 
 
+Published Workspaces
+#######################
```
Comment (abhilash-kumar-nair) [inline docs/source/working_with_workspaces.rst:67]: Could you remove the extra hash?
Published Workspaces
##################

File: docs/source/working_with_workspaces.rst [hunk 3]
```diff
@@ -57,9 +57,48 @@ would result in multiple workspaces being created with same name.
 Workspace sharing
 *****************
 
-A workspace in Modelon Impact can be shared as a zipped artifact(snapshot) or as workspace definition referring to version controlled 
+A workspace in Modelon Impact can be shared as a zipped artifact (snapshot) or as a workspace configuration.
+A snapshot can be either stored in a cloud storage and shared with the other users of the same 
+Modelon Impact site or downloaded as a zip file to the client system.
```
Comment (abhilash-kumar-nair) [inline docs/source/working_with_workspaces.rst:62]: Modelon Impact site - A bit ambiguous on what site means here. Group or website. Not sure if adding a section about tenants/groups and then use Modelon Impact group makes sence?

**Review by abhilash-kumar-nair — COMMENTED** [review]
Looks neat :100: !  Just a few minor comments. :)

**Comment by iakovn** [comment]
Thanks for the review! Fixed/clarified. Please, have another look

**Review by abhilash-kumar-nair — APPROVED** [review]
Looks good! :100:  Lets get this in!

**Pull request merged**